### PR TITLE
AGNT-586 Add support for simplified key delivery

### PR DESCRIPTION
**Pull request by FabienVSymphony**
### Description
Closes #[AGNT-585]

When simplified key delivery is enabled and agent is running a version >= 24.12 we stop retrieving the keymanager token. This allows the user to run bots on cloud where an on premise key manager is not available

### Dependencies
We depend on https://github.com/finos/symphony-api-spec/pull/199 being merged before we can merge this PR as we use the swagger definitions to generate our APIs

### CODE REVIEW

**Review by vladokrsymphony — APPROVED** [review]

**Review by tzhao-symphony — APPROVED** [review]

File: symphony-bdk-core/build.gradle [hunk 1]
```diff
@@ -75,7 +75,7 @@ dependencies {
 }
 
 // OpenAPI code generation
-def apiBaseUrl = "https://raw.githubusercontent.com/finos/symphony-api-spec/30fcab0fe6eaa26dcc46e7dc5909467332ec8d0d"
+def apiBaseUrl = "https://raw.githubusercontent.com/FabienVSymphony/symphony-api-spec/refs/heads/AGNT-585/"
```
Comment (yinan-symphony) [inline symphony-bdk-core/build.gradle:78]: this should be finos link

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/version/AgentVersionService.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import com.symphony.bdk.http.api.ApiException;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apiguardian.api.API;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
```
Comment (yinan-symphony) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/version/AgentVersionService.java:15]: java doc pls

File: symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/version/model/AgentVersion.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.symphony.bdk.core.service.version.model;
+
+import org.apiguardian.api.API;
+
+@API(status = API.Status.INTERNAL)
```
Comment (yinan-symphony) [inline symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/version/model/AgentVersion.java:5]: java doc pls

**Review by yinan-symphony — APPROVED** [review]
by adding java docs on the two new classes, we are good.

**Pull request merged**